src/ess/reduce/scicat.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+from typing import Optional
+
+from scitacean import Client
+from scitacean.transfer.sftp import SFTPFileTransfer
+
+from .nexus import FilePath
+
+
+def download_scicat_file(
+    dataset_id: str,
+    filename: str,
+    *,
+    token: str,
+    version: Optional[str] = None,
```

Comment:
Better parametrize over the client. This allows users to construct their own if they have special needs (e.g., special SSH auth) and we can use a fake in tests.